docs/migrate.ipynb

```diff
@@ -254,7 +254,45 @@
       "source": [
         "## Converting your code\n",
         "\n",
-        "Converting existing `tf.summary` usage to the TF 2.x API cannot be reliably automated, so the [`tf_upgrade_v2` script](https://www.tensorflow.org/guide/upgrade) just rewrites it all to `tf.compat.v1.summary`.  To migrate to TF 2.x, you'll need to adapt your code as follows:"
+        "Converting existing `tf.summary` usage to the TF 2.x API cannot be reliably automated, so the [`tf_upgrade_v2` script](https://www.tensorflow.org/guide/upgrade) just rewrites it all to `tf.compat.v1.summary` and will not enable the TF 2.x behaviors automatically."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1972f8ff0073"
+      },
+      "source": [
+        "### Partial Migration\n",
+        "\n",
+        "It is possible to invoke TF 2.x behaviors within <a href=\"https://www.tensorflow.org/api_docs/python/tf/compat/v1/summary\"><code>tf.compat.v1.summary</code></a> APIs under the following conditions:\n",
+        "\n",
+        " - The outermost context is eager mode\n",
+        " - A suitable TF2 summary writer exists\n",
+        " - A non-empty value for step is set for the writer (using <a href=\"https://www.tensorflow.org/api_docs/python/tf/summary/SummaryWriter#as_default\"><code>tf.summary.SummaryWriter.as_default</code></a>, or alternatively <a href=\"https://www.tensorflow.org/api_docs/python/tf/summary/experimental/set_step\"><code>tf.summary.experimental.set_step</code></a>)\n",
```

Comment:
We could also mention `tf.compat.v1.train.*_global_step()` here too, since it does also work. Up to you though (I realize we left it out for the API docs).

Reply:
Thanks for the reminder. In my mind 💭 for some reason I felt that it was updated in the API doc string, but it's not the case in reality. Will also update the API docs later.